### PR TITLE
Add missing kdoc

### LINKED ISSUE #155: Add documentation to all public classes, methods and fields
As required from the code review. Will make everything easier to read.
Use https://kotlinlang.org/docs/kotlin-doc.html

### PR DESCRIPTION
**Pull request by wattsecs**

### CODE REVIEW

File: app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/hilt.kt [hunk 1]
```diff
@@ -25,7 +25,7 @@ class FirebaseFirestoreTestModule {
     @Singleton
     fun provideFirestore(): FirebaseFirestore {
         FirebaseFirestore.getInstance()
-            .terminate() //TODO("Find out why it is initialized before this instead of just terminating it before restarting")
```
Comment (bbjubjub2494) [inline app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/hilt.kt]: Do we not want TODO that anymore?

File: app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/view/GoogleSignInFragmentTest.kt [hunk 2]
```diff
@@ -26,12 +26,12 @@ import org.junit.runner.RunWith
 
 @HiltAndroidTest
 @RunWith(AndroidJUnit4::class)
-class GoogleSignInWidgetTest {
```
Comment (bbjubjub2494) [inline app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/view/GoogleSignInFragmentTest.kt:29]: :+1:

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/UniqueDatabaseManagement.kt [hunk 3]
```diff
@@ -131,6 +161,10 @@ class UniqueDatabaseManagement @Inject constructor(
             DefaultDatabaseManagement(OfflineDatabaseService(databaseName, context, room))
         }
 
+    /**
+     * Removes a database from on device storage asynchronously.
+     * @param databaseName of the database to remove from on device storage
+     */
     fun removeDatabaseFromDownloadsAsync(databaseName: String) =
```
Comment (bbjubjub2494) [inline app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/UniqueDatabaseManagement.kt]:
```suggestion
     * @return a job tracking completion of the operation
     */
    fun removeDatabaseFromDownloadsAsync(databaseName: String): Job =
```

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/room/Daos.kt [hunk 4]
```diff
@@ -81,6 +87,9 @@ interface DatasetDao {
     suspend fun delete(vararg refs: RoomDatasetCategoriesCrossRef)
 }
 
+/**
+ * Data accessor object for everything concerning databases in on device storage
```
Comment (bbjubjub2494) [inline app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/room/Daos.kt]:
```suggestion
 * Data access object for everything concerning databases in on device storage
```

**Review by bbjubjub2494 — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/hilt.kt (hunk 1 above)
Comment (wattsecs) [reply app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/hilt.kt]: Huh... I readded it

**Review by Jaakik — APPROVED** [review]
All good.  Just a reminder, I have added some Kdoc for the signIn Fragment

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/UniqueDatabaseManagement.kt (hunk 3 above)
Comment (bbjubjub2494) [reply app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/UniqueDatabaseManagement.kt]: please address this

**Review by bbjubjub2494 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/UniqueDatabaseManagement.kt (hunk 3 above)
Comment (wattsecs) [reply app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/UniqueDatabaseManagement.kt]: I missed it sorry

**Review by bbjubjub2494 — APPROVED** [review]
LGTM. Just make sure not to re-add Statistic

**Pull request merged**